enclave/src/polkadex_gateway.rs

```diff
@@ -186,24 +253,56 @@ fn send_cancel_request_to_openfinex(order_uuid: &OrderUUID) -> Result<(), Gatewa
 pub fn cancel_order(
     main_account: AccountId,
     proxy_acc: Option<AccountId>,
-    order_uuid: OrderUUID,
+    cancel_order: CancelOrder,
 ) -> Result<(), GatewayError> {
     // Authenticate
     authenticate_user(main_account.clone(), proxy_acc)?;
-    // if let Ok(mutex) = load_cancel_cache_pointer() {
-    //     let mut cancel_cache: SgxMutexGuard<HashSet<OrderUUID>> = mutex.lock().unwrap();
-    //     cancel_cache.insert(order_uuid);
-    //     // Send cancel order to Openfinex API
-    // }
-    send_cancel_request_to_openfinex(&order_uuid)?;
+    let mutex = CancelOrderCache::load().map_err(|_| GatewayError::NullPointer)?;
+    let mut cache = match mutex.lock() {
+        Ok(guard) => guard,
+        Err(e) => {
+            error!("Could not acquire lock on cancel cache pointer: {}", e);
+            return Err(GatewayError::UnableToLock);
+        }
+    };
+
+    send_cancel_request_to_openfinex(cancel_order.clone(), cache.request_id() as RequestId)?;
+    cache.insert_order(cancel_order.order_id);
+
+    Ok(())
+}
+
+// /// process_cancel_order does the following
+// /// 1. Checks the orderUUID with cancel request cache
+// /// 2. Remove order from Orderbook Mirror
+// /// 3. Mutate the balances
+pub fn process_cancel_order(order_uuid: OrderUUID) -> Result<(), GatewayError> {
+    let mutex = CancelOrderCache::load().map_err(|_| GatewayError::NullPointer)?;
+    let mut cancel_cache = match mutex.lock() {
+        Ok(guard) => guard,
+        Err(e) => {
+            error!("Could not acquire lock on cancel cache pointer: {}", e);
+            return Err(GatewayError::UnableToLock);
+        }
+    };
+
+    if !cancel_cache.remove_order(&order_uuid) {
+        error!("Order Cancel Request not found in Cache");
+        return Err(GatewayError::UnableToCancelOrder);
+    }
+
     let cancelled_order = polkadex_orderbook_storage::lock_storage_and_remove_order(&order_uuid)?;
     match (cancelled_order.order_type, cancelled_order.side) {
         (OrderType::LIMIT, OrderSide::BID) => {
             let price = cancelled_order
                 .price
                 .ok_or(GatewayError::LimitOrderPriceNotFound)?;
+
+            // TODO: Why are we converting the amounts here? The OpenFinex API is responsible for doing all the conversions?
+            // Even if we had to convert, we should not do a floating point conversion
             let amount =
```

Comment:
yes, we shouldn't use the float here, we can change it to substrate-fixed

Reply:
@mullefel I think you are working on the fixed point numbers once your PR is merged we can change this one